File: 3869-smallest-index-with-digit-sum-equal-to-index/smallest-index-with-digit-sum-equal-to-index.py
class Solution:
    def smallestIndex(self, nums: List[int]) -> int:

        def digit(n):
            total = 0

            while n > 0:
                d = n % 10
                total += d
                n //= 10

            return total

        index = float('inf')

        for i, num in enumerate(nums):
            if i == digit(num):
                index = min(index, i)

        return -1 if index == float('inf') else index

File: 3869-smallest-index-with-digit-sum-equal-to-index/smallest-index-with-digit-sum-equal-to-index.java
class Solution {

    private int digitsum(int n){
        int total = 0;
        while (n > 0){
            int digit = n % 10;
            total = total+digit;
            n /= 10;
        }
        return total;
    }
    public int smallestIndex(int[] nums) {
        int n = nums.length;
        int index = Integer.MAX_VALUE;
        for(int i = 0; i < n; i++){
            if (i == digitsum(nums[i])){
                index = Math.min(index, i);
            }
        }

        return index == Integer.MAX_VALUE? -1 : index;
    }
}